.github/workflows/push_release.yml

```diff
@@ -0,0 +1,93 @@
+name: Push Main to Release Branch Sync
+
+on:
+  workflow_dispatch:
+    inputs:
+      org:
+        description: 'Select Organization'
+        required: true
+        type: choice
+        options:
+          - 'opendatahub.io'
+          - 'red-hat-data-services'
+      sync_repo:
+        description: 'Select Repository'
+        required: true
+        type: choice
+        options:
+          - 'modelmesh'
+          - 'caikit-tgis-serving'
+          - 'openvino'
+          - 'vllm'
+          - 'caikit-nlp'
+          - 'caikit'
+          - 'odh-model-controller'
+          - 'caikit-tgis-backend'
+          - 'caikit-nlp-client'
+          - 'model-registry'
+      main_branch:
+        description: 'Main branch to pull from'
+        default: 'main'
+        required: true
+      release_branch:
+        description: 'Branch to push to [release branch]'
+        required: true
+
+env:
+  ACTIONS_PAT: ${{ secrets.ACTIONS_PAT }}
+
+permissions:
+  contents: write
+  
+
+jobs:
+  push-release-branch:
+    runs-on: ubuntu-latest
+    steps:
+
+      - name: Set up Git
+        run: |
+          git config --global user.name 'github-actions'
+          git config --global user.email 'github-actions@github.com'
+
+      - name: Checkout repository
+        uses: actions/checkout@v4
+        with:
+            repository: ${{ github.event.inputs.org }}/${{ github.event.inputs.sync_repo }}
+            token: ${{ secrets.ACTIONS_PAT }}
+            fetch-depth: 0
+
+      
+      - name: Print current repository
+        run: git remote -v
+
+      - name: Fetch all branches
+        run: git fetch --all
+
+
+      - name: Merge Master to Production branch
+        uses: devvspaces/merge-branches@v1
+        with:
+              token: ${{ secrets.ACTIONS_PAT }}
+              from_branch: ${{ github.event.inputs.main_branch }}
+              to_branch: ${{ github.event.inputs.release_branch }}
```

Comment:
As per my understanding these lines of code will directly push the code (from main --- to --- release branch). Thus, the lines of code below are redundant.

But as per the goal of the task we need to sync main to relase branch via PR(not with direct push). I think you need to firstly checkout to new branch from target branch (i.e., release branch ) then push the changes of main to that branch (i.e., new branch) and then create a PR from that branch(new branch) to the release/target branch .